src/app/qgisapp.cpp

```diff
@@ -12840,31 +12842,36 @@ bool QgisApp::checkMemoryLayers()
     }
     else if ( it.value() && it.value()->isTemporary() )
       hasTemporaryLayers = true;
+    else if ( it.value() && it.value()->type() == QgsMapLayerType::MeshLayer )
+    {
+      QgsMeshLayer *meshLayer = qobject_cast< QgsMeshLayer * >( it.value() );
+      if ( meshLayer && meshLayer->memoryDatasetGroupCount() > 0 )
+      {
+        hasMemoryMeshLayerDatasetGroup = true;
+      }
+    }
   }
 
+  bool close = true;
   if ( hasTemporaryLayers )
-  {
-    if ( QMessageBox::warning( this,
-                               tr( "Close Project" ),
-                               tr( "This project includes one or more temporary layers. These layers are not permanently saved and their contents will be lost. Are you sure you want to proceed?" ),
-                               QMessageBox::Yes | QMessageBox::Cancel, QMessageBox::Cancel ) == QMessageBox::Yes )
-      return true;
-    else
-      return false;
-  }
+    close &= QMessageBox::warning( this,
+                                   tr( "Close Project" ),
+                                   tr( "This project includes one or more temporary layers. These layers are not permanently saved and their contents will be lost. Are you sure you want to proceed?" ),
```

Comment:
This project -> The project

Reply:
I keep the sentence as other pre existing warning just above.